libconfluo/test/atomic_multilog_test.h

```diff
@@ -76,6 +79,30 @@ class AtomicMultilogTest : public testing::Test {
     return reinterpret_cast<void *>(&r);
   }
 
+  static std::string make_json_record(int64_t ts, bool a, int8_t b, int16_t c, int32_t d, int64_t e, float f, double g, const std::string &h) {
+    namespace pt = boost::property_tree;
+    pt::ptree root;
+
+    root.put("TIMESTAMP", ts);
+    root.put("A", a);
+    root.put("B", b);
+    root.put("C", c);
+    root.put("D", d);
+    root.put("E", e);
+    std::stringstream f_ss;
+    f_ss << std::fixed << std::setprecision(6) << f;
+    root.put("F", f_ss.str());
+    std::stringstream g_ss;
+    g_ss << std::fixed << std::setprecision(6) << g;
+    root.put("G", g_ss.str());
+    root.put("H", h);
+
+    std::stringstream ss;
+    pt::write_json(ss, root);
+    std::string ret = ss.str();
```

Comment:
Maybe try printing this out and use the printed string as your test input? Tests are also useful as sample usage for users.